modules/ballerina-lang/src/main/java/org/ballerinalang/spi/UserRepositoryProvider.java

```diff
@@ -0,0 +1,38 @@
+/*
+*  Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+*
+*  WSO2 Inc. licenses this file to you under the Apache License,
+*  Version 2.0 (the "License"); you may not use this file except
+*  in compliance with the License.
+*  You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+*  Unless required by applicable law or agreed to in writing,
+*  software distributed under the License is distributed on an
+*  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+*  KIND, either express or implied.  See the License for the
+*  specific language governing permissions and limitations
+*  under the License.
+*/
+package org.ballerinalang.spi;
+
+import org.ballerinalang.repository.PackageRepository;
+
+/**
+ * This represents the Java SPI interface for a Ballerina user package repository provider.
+ * It has highest presidency, over system and extension repos.
+ *
+ * @since 0.94
+ */
+public interface UserRepositoryProvider {
```

Comment:
Creating an interface for an user repository provider concept and using Java SPI for this is not essential, since the user repository implementation is fixed, and is not to be intended to be extended later with multiple providers. If there's a possibility of such a requirement may come, then we can keep this.

Reply:
I did it this way to avoid maven circular dependency, because `UserRepository` is in packerina and `PackageLoader` is in lang. Right now only packerina deals with '~/.ballerina' dir. Other option is to move that logic to lang, but I think keeping it in packerina is better.